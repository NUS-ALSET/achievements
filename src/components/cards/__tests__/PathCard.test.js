import React from "react";
import { createShallow } from "@material-ui/core/test-utils";
import Typography from "@material-ui/core/Typography";
import PathCard from "../PathCard";

describe("<PathCard />", () => {
  let shallow;
  beforeEach(() => {
    shallow = createShallow();
  });

  it("Should test Typography into PathCard component", () => {
    const component = shallow(<PathCard title={"test"} />);
    const typographys = component.find(Typography);
    expect(typographys.length).toEqual(1);
  });
});
