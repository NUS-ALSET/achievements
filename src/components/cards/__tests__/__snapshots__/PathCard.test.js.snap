// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PathCard /> Should test PathCard component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PathCard
    title=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        undefined,
        <WithStyles(CardContent)>
          <WithStyles(Typography) />
</WithStyles(CardContent)>,
        <WithStyles(CardActions)
          style={
                    Object {
                              "background": "rgba(255,255,255,0.54)",
                              "bottom": 0,
                              "justifyContent": "flex-end",
                              "position": "absolute",
                              "width": "100%",
                            }
          }
>
          <Link
                    replace={false}
                    to="/paths/undefined/problems/undefined"
          >
                    <WithStyles(IconButton)
                              color="inherit"
                    >
                              <pure(KeyboardArrowRight) />
                    </WithStyles(IconButton)>
          </Link>
</WithStyles(CardActions)>,
      ],
      "style": Object {
        "flex": 1,
        "margin": 4,
        "minHeight": 120,
        "minWidth": 160,
        "position": "relative",
        "width": "25%",
      },
    },
    "ref": null,
    "rendered": Array [
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithStyles(Typography) />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Link
            replace={false}
            to="/paths/undefined/problems/undefined"
>
            <WithStyles(IconButton)
                        color="inherit"
            >
                        <pure(KeyboardArrowRight) />
            </WithStyles(IconButton)>
</Link>,
          "style": Object {
            "background": "rgba(255,255,255,0.54)",
            "bottom": 0,
            "justifyContent": "flex-end",
            "position": "absolute",
            "width": "100%",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(IconButton)
              color="inherit"
>
              <pure(KeyboardArrowRight) />
</WithStyles(IconButton)>,
            "replace": false,
            "to": "/paths/undefined/problems/undefined",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <pure(KeyboardArrowRight) />,
              "color": "inherit",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          undefined,
          <WithStyles(CardContent)>
            <WithStyles(Typography) />
</WithStyles(CardContent)>,
          <WithStyles(CardActions)
            style={
                        Object {
                                    "background": "rgba(255,255,255,0.54)",
                                    "bottom": 0,
                                    "justifyContent": "flex-end",
                                    "position": "absolute",
                                    "width": "100%",
                                  }
            }
>
            <Link
                        replace={false}
                        to="/paths/undefined/problems/undefined"
            >
                        <WithStyles(IconButton)
                                    color="inherit"
                        >
                                    <pure(KeyboardArrowRight) />
                        </WithStyles(IconButton)>
            </Link>
</WithStyles(CardActions)>,
        ],
        "style": Object {
          "flex": 1,
          "margin": 4,
          "minHeight": 120,
          "minWidth": 160,
          "position": "relative",
          "width": "25%",
        },
      },
      "ref": null,
      "rendered": Array [
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Typography) />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "",
            },
            "ref": null,
            "rendered": "",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Link
              replace={false}
              to="/paths/undefined/problems/undefined"
>
              <WithStyles(IconButton)
                            color="inherit"
              >
                            <pure(KeyboardArrowRight) />
              </WithStyles(IconButton)>
</Link>,
            "style": Object {
              "background": "rgba(255,255,255,0.54)",
              "bottom": 0,
              "justifyContent": "flex-end",
              "position": "absolute",
              "width": "100%",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(IconButton)
                color="inherit"
>
                <pure(KeyboardArrowRight) />
</WithStyles(IconButton)>,
              "replace": false,
              "to": "/paths/undefined/problems/undefined",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <pure(KeyboardArrowRight) />,
                "color": "inherit",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {},
  },
}
`;
